src/client/src/orangehrmClaimPlugin/pages/ClaimEvent.vue

```diff
@@ -0,0 +1,256 @@
+</template>
+
+<script>
+import {ref, computed} from 'vue';
+import {APIService} from '@/core/util/services/api.service';
+import usePaginate from '@ohrm/core/util/composable/usePaginate';
+import useSort from '@ohrm/core/util/composable/useSort';
+import {navigate} from '@/core/util/helper/navigation';
+import DeleteConfirmationDialog from '@ohrm/components/dialogs/DeleteConfirmationDialog.vue';
+
+const defaultFilters = {
+  name: '',
+  status: null,
+};
+
+const defaultSortOrder = {
+  'claimEvent.name': 'ASC',
+  'claimEvent.status': 'DESC',
+};
+
+export default {
+  components: {
+    'delete-confirmation': DeleteConfirmationDialog,
+  },
+  setup() {
+    const filters = ref({...defaultFilters});
+
+    const {sortDefinition, sortField, sortOrder, onSort} = useSort({
+      sortDefinition: defaultSortOrder,
+    });
+
+    const serializedFilters = computed(() => {
+      return {
+        name: filters.value.name,
+        status: filters.value.status ? filters.value.status?.id === 1 : null,
+        sortField: sortField.value,
+        sortOrder: sortOrder.value,
+      };
+    });
+
+    const http = new APIService(
+      window.appGlobal.baseUrl,
+      'api/v2/claim/events',
+    );
+    const {
+      showPaginator,
+      currentPage,
+      total,
+      pages,
+      pageSize,
+      response,
+      isLoading,
+      execQuery,
+    } = usePaginate(http, {query: serializedFilters});
+    onSort(execQuery);
+
+    return {
+      http,
+      showPaginator,
+      currentPage,
+      isLoading,
+      total,
+      pages,
+      pageSize,
+      execQuery,
+      items: response,
+      response,
+      filters,
+      sortDefinition,
+    };
+  },
+  data() {
+    return {
+      headers: [
+        {
+          name: 'name',
+          title: this.$t('general.name'),
+          slot: 'title',
+          sortField: 'claimEvent.name',
+          style: {flex: 3},
+        },
+        {
+          name: 'status',
+          title: 'Status',
+          sortField: this.$t('general.status'),
+          style: {flex: 2},
+        },
+        {
+          name: 'actions',
+          title: this.$t('general.actions'),
+          slot: 'action',
+          style: {flex: 1},
+          cellType: 'oxd-table-cell-actions',
+          cellConfig: {
+            delete: {
+              onClick: () => {
+                this.onClickDelete();
+              },
+              component: 'oxd-icon-button',
+              props: {
+                name: 'trash',
+              },
+            },
+            edit: {
+              onClick: () => {
+                return;
+              },
+              props: {
+                name: 'pencil-fill',
+              },
+            },
+          },
+        },
+      ],
+      checkedItems: [],
+      ClaimEventStatuses: [
+        {id: 1, label: this.$t('Active')},
+        {id: 0, label: this.$t('Inactive')},
+      ],
+    };
+  },
+
+  computed: {
+    items_new() {
```

Comment:
you can use a `normalizer` function as a parameter for `usePaginate` function. `normalizer` is a function that can transform raw data retrived via API before getting output.

https://github.com/orangehrm/orangehrm/blob/main/src/client/src/orangehrmAdminPlugin/pages/systemUser/SystemUser.vue#L200